selfservice/flow/recovery/flow.go

```diff
@@ -147,7 +150,7 @@ func NewFlow(conf *config.Config, exp time.Duration, csrf string, r *http.Reques
 func FromOldFlow(conf *config.Config, exp time.Duration, csrf string, r *http.Request, strategy Strategy, of Flow) (*Flow, error) {
 	f := of.Type
 	// Using the same flow in the recovery/verification context can lead to using API flow in a verification/recovery email
-	if of.Type == flow.TypeAPI {
+	if of.Type == flow.TypeAPI && of.Active.String() == string(RecoveryStrategyLink) {
```

Comment:
nit: you probably don't need to convert both to string

Reply:
`of.Active` is a `sqlxx.NullString`: 

<img width="951" alt="image" src="https://github.com/ory/kratos/assets/5540255/5e39cdf8-1680-4725-87ce-615a5ca94496">